RecoTracker/LSTCore/interface/Constants.h

```diff
@@ -43,15 +43,15 @@ namespace lst {
   constexpr unsigned int max_blocks = 80;
   constexpr unsigned int max_connected_modules = 40;
 
-  constexpr unsigned int n_max_pixel_segments_per_module = 50000;
+  constexpr unsigned int n_max_pixel_segments_per_module = 500000;
 
   constexpr unsigned int n_max_pixel_md_per_modules = 2 * n_max_pixel_segments_per_module;
 
   constexpr unsigned int n_max_pixel_triplets = 5000;
   constexpr unsigned int n_max_pixel_quintuplets = 15000;
 
-  constexpr unsigned int n_max_pixel_track_candidates = 30000;
-  constexpr unsigned int n_max_nonpixel_track_candidates = 1000;
+  constexpr unsigned int n_max_pixel_track_candidates = 3000000;
+  constexpr unsigned int n_max_nonpixel_track_candidates = 100000;
```

Comment:
what's the cost to the memory used from changes in this file?

Reply:
I am not sure.

In the final PR, these increase "only" by 10x (and I can probably fine-tune it to something less). They are also needed only in the case of `seedingLST`, which I would not use at this stage. Maybe I should not touch them and revisit if/when the `seedingLST` config becomes usable?

Reply:
I'm partly confused what "final PR" means. Does it mean that at least this PR target branch should be updated or the PR itself to be rebased?

I think https://docs.google.com/spreadsheets/d/1-tDK4nIPlrB-0SoeEbnJlf_lyt-xTZcIhmZGRnR-fnY/edit?gid=16484832#gid=16484832 is still mostly up to date at least for memory per object. So, just the expected memory use would be enough to clarify my question.

Reply:
> I'm partly confused what "final PR" means. Does it mean that at least this PR target branch should be updated or the PR itself to be rebased?

Also the PR needs to be rebased. Small adjustments but it needs to. The target branch will also change - should I just switch to `SegmentLinking/master`, after I sync it with the upstream, or will that cause issues?

> I think https://docs.google.com/spreadsheets/d/1-tDK4nIPlrB-0SoeEbnJlf_lyt-xTZcIhmZGRnR-fnY/edit?gid=16484832#gid=16484832 is still mostly up to date at least for memory per object. So, just the expected memory use would be enough to clarify my question.

The spreadsheet was not up to date. I updated the default numbers and I get:
- size = 1208.8
- cache size = 1503.8

When using the values for the upcoming PR (the latter two 10x less than what is there in this PR):
```
n_max_pixel_segments_per_module = 500000;
n_max_pixel_track_candidates = 300000;
n_max_nonpixel_track_candidates = 10000;
```
I get:
- size = 1274.6
- cache size = 1581.1

I.e. ~5% increase in size and cache size. I will post the same numbers in PR description of the new PR.

Reply:
> The target branch will also change - should I just switch to `SegmentLinking/master`, after I sync it with the upstream, or will that cause issues?

master should be OK as a target, but please start the topic branch from some pre-release (pre4?) if possible. This way it should be easy to add it also to an LST development branch once we converge on the name and merge-base.

Reply:
Indeed, I started from `CMSSW_14_2_0_pre4` (made it clear also in the branch name and the PR description).